contributors/design-proposals/storage/csi-migration.md

```diff
@@ -35,12 +35,28 @@ lower cost of development as we only have to maintain one version of each
 plugin, as well as ease the transition to CSI when we are able to deprecate the
 internal APIs.
 
+The migration from in-tree plugins to CSI plugins will involve the following phases:
+
+Phase 1: In-tree plugins are probed but the in-tree plugin code is largely skipped
+for operations on persistent volumes in favor of CSI plugins that supersede the
+in-tree plugins. In case a node does not have a CSI plugin (that supersedes a
+given in-tree plugin) installed and configured, the in-tree plugin code
+will be executed for specific operations like attachment/detachment and attachment
+verification of volumes.
+
+Phase 2: In-tree plugins are no longer probed and thus in-tree code is effectively
+disabled (but still compiled). All nodes in a cluster will be expected to have CSI
+plugins (that supersede the disabled in-tree plugins) installed and configured.
+
+Phase 3: Files containing in-tree plugin code are no longer compiled as part of
+Kubernetes components using golang build tags introduced in
+https://github.com/kubernetes/kubernetes/pull/80353
 
 ## Goals
 
 * Compile all requirements for a successful transition of the in-tree plugins to
   CSI
-    * As little code as possible remains in the Kubernetes Repo
+    * In-tree plugin code can be completely disabled in preparation for removal
```

Comment:
could you modify wording here, since "removing it" is the end goal of migration, not "preparation for removal"